tuned/daemon/controller.py

```diff
@@ -417,3 +417,74 @@ def instance_get_devices(self, instance_name, caller = None):
 		rets = "Instance '%s' not found" % instance_name
 		log.error(rets)
 		return (False, rets, [])
+
+	@exports.export("ssa{ss}", "bs")
+	def instance_create(self, plugin_name, instance_name, options, caller = None):
+		"""Dynamically create a plugin instance
+
+		Parameters:
+		plugin_name -- name of the plugin
+		instance_name -- name of the new instance
+		dict of string-string -- options for the new instance
+
+		Return:
+		bool -- True on success
+		string -- error message or "OK"
+		"""
+		plugins = {p.name: p for p in self._daemon._unit_manager.plugins}
+		if not plugin_name in plugins.keys():
+			rets = "Plugin '%s' not found" % plugin_name
+			log.error(rets)
+			return (False, rets)
+		plugin = plugins[plugin_name]
+		instance = plugin.create_instance(instance_name, None, None, None, None, options)
+		plugin.initialize_instance(instance)
+		instance._created_dynamically = True
+		self._daemon._unit_manager.instances.append(instance)
+		log.info("Created dynamic instance '%s' of plugin '%s'" % (instance_name, plugin_name))
+		return (True, "OK")
+
+	@exports.export("ss", "bs")
+	def instance_destroy(self, plugin_name, instance_name, caller = None):
+		"""Destroy a dynamically created plugin instance
+
+		Parameters:
+		plugin_name -- name of the plugin
+		instance_name -- name of the new instance
+
+		Return:
+		bool -- True on success
+		string -- error message or "OK"
+		"""
+		instance = None
+		for i in self._daemon._unit_manager.instances:
+			if i.name == instance_name:
+				instance = i
+				break
+		if instance is None:
+			rets = "Instance '%s' not found" % instance_name
+			log.error(rets)
+			return (False, rets)
+		if not instance._created_dynamically:
+			rets = "Cannot delete instance '%s' because it was not created dynamically" % instance_name
+			log.error(rets)
+			return (False, rets)
+		if not instance.plugin.name == plugin_name:
+			rets = "Instance '%s' does not belong to plugin '%s'" % (instance_name, plugin_name)
+			log.error(rets)
+			return (False, rets)
+		if len(instance._processed_devices) > 0:
+			rets = "Cannot delete instance '%s' because it still has attached devices" % instance_name
```

Comment:
For simple PoC it's probably OK, but maybe you could free the devices and try to hotplug them (if possible).

Reply:
#596 now implements this.